stglib/iq.py

```diff
@@ -91,142 +96,174 @@ def read_iq(filnam):
-        "AdjustedPressure": "P_1ac",
+    meanveleq = {
+        "1": "Theoretical",
+        "2": "Index",
+        "3": "Theoretical (Center Beams Only)",
+        "4": "Theoretical (Beam 1 Only)",
+        "5": "Theoretical (Beam 2 Only)",
     }
 
-    # check to make sure they exist before trying to rename
-    newvars = {}
-    for k in varnames:
-        if k in ds:
-            newvars[k] = varnames[k]
-
-    return ds.rename(newvars)
-
-
-def remove_FlowData(ds):
-    newvars = {}
-    for k in ds:
-        newvars[k] = k.replace("FlowData_", "")
-
-    return ds.rename(newvars)
+    for v in meanveleq:
+        if v in ds.attrs["mean_velocity_equation_type"]:
+            ds.attrs["mean_velocity_equation_type"] = meanveleq[v]
 
+    if ds.attrs["mean_velocity_equation_type"] == "Index":
+        ds.attrs["equation_velocity_type"] = str(
+            iqmat["System_IqSetup"]["flowSetup"]["velocityType"]
+        )
 
-def clean_iq(iq):
-    """
-    Preliminary data cleaning when SNR < 0
-    """
-
-    iq["Vel_Mean"].values[iq["Vel_Mean"] < -214748] = np.nan
-    iq["Vel"].values[iq["Vel"] == -214748368] = np.nan
-    for bm in range(4):
-        pr = "Profile_" + str(bm) + "_Vel"
-        iq[pr].values[iq[pr] == -214748368] = np.nan
-        am = "Profile_" + str(bm) + "_Amp"
-        iq[am].values[iq[am] == 65535] = np.nan
-        st = "Profile_" + str(bm) + "_VelStd"
-        iq[st].values[iq[st] < 0] = np.nan
-
-    return iq
-
+    if "equation_velocity_type" in ds.attrs:
+        veltype = {
+            "0": "VelocityXZ (X-Center)",
+            "2": "VelocityXZ (Z-Center)",
+            "3": "VelocityXZ (X-Left)",
+            "4": "VelocityXZ (X-Right)",
+            "5": "Average of all Vx",
+            "10": "Beam 1 Velocity",
+            "11": "Beam 2 Velocity",
+            "12": "Beam 3 Velocity",
+            "13": "Beam 4 Velocity",
+            "14": "VelocityXZ (X-Center, Beam 1 Only)",
+            "15": "VelocityXZ (X-Center, Beam 2 Only)",
+        }
+
+        for v in veltype:
+            if v in ds.attrs["equation_velocity_type"]:
+                ds.attrs["equation_velocity_type"] = veltype[v]
+
+    ds.attrs["survey_point_count"] = int(
+        iqmat["System_IqSetup"]["flowSetup"]["surveyPointCount"]
+    )
+    ds.attrs["survey_point_count"] = int(
+        iqmat["System_IqSetup"]["flowSetup"]["surveyPointCount"]
+    )
+    ds.attrs["IQ_location_Y"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["instrument_Y"] / 1000
+    )
+    ds.attrs["IQ_location_Z"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["instrument_Z"] / 1000
+    )
+    ds.attrs["channel_cross_section_Y"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["channel_Y"][
+            0 : ds.attrs["survey_point_count"]
+        ]
+    ) / 1000
+    ds.attrs["channel_cross_section_Z"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["channel_Z"][
+            0 : ds.attrs["survey_point_count"]
+        ]
+    ) / 1000
+
+    ds.attrs[
```

Comment:
This feels like a note that might not be the case for every IQ deployment. Should it be added in the config file instead?

Reply:
I agree. I think in general the comment fits well for the IQ deployments but specifying the starting bank will change. I can take this out.